crates/forge_app/src/services.rs

```diff
@@ -841,26 +852,37 @@ pub trait HttpClientService: Send + Sync + 'static {
     ) -> anyhow::Result<EventSource>;
 }
 
+#[async_trait::async_trait]
+impl<I: Services> AgentLoader for I {
+    async fn get_agents(&self) -> anyhow::Result<Vec<forge_domain::AgentDefinition>> {
+        self.agent_loader().get_agents().await
+    }
+}
+
 #[async_trait::async_trait]
 impl<I: Services> AgentRegistry for I {
-    async fn get_agents(&self) -> anyhow::Result<Vec<Agent>> {
-        self.agent_registry().get_agents().await
+    async fn get_active_agent_id(&self) -> anyhow::Result<Option<AgentId>> {
+        self.agent_registry().get_active_agent_id().await
     }
 
-    async fn get_agent(&self, agent_id: &AgentId) -> anyhow::Result<Option<Agent>> {
-        self.agent_registry().get_agent(agent_id).await
+    async fn set_active_agent_id(&self, agent_id: AgentId) -> anyhow::Result<()> {
+        self.agent_registry().set_active_agent_id(agent_id).await
+    }
+
+    async fn get_agents(&self) -> anyhow::Result<Vec<forge_domain::Agent>> {
+        self.agent_registry().get_agents().await
     }
 
-    async fn get_active_agent(&self) -> anyhow::Result<Option<Agent>> {
-        self.agent_registry().get_active_agent().await
+    async fn get_agent(&self, agent_id: &AgentId) -> anyhow::Result<Option<forge_domain::Agent>> {
+        self.agent_registry().get_agent(agent_id).await
     }
 
-    async fn get_active_agent_id(&self) -> anyhow::Result<Option<AgentId>> {
-        self.agent_registry().get_active_agent_id().await
+    async fn set_agents(&self, agents: Vec<forge_domain::Agent>) -> anyhow::Result<()> {
+        self.agent_registry().set_agents(agents).await
     }
 
-    async fn set_active_agent_id(&self, agent_id: AgentId) -> anyhow::Result<()> {
-        self.agent_registry().set_active_agent_id(agent_id).await
+    async fn insert_agent(&self, agent: forge_domain::Agent) -> anyhow::Result<()> {
+        self.agent_registry().insert_agent(agent).await
     }
```

Comment:
```suggestion    
```

Use the AgentRepo in Registry to fetch and cache AgentDef to Agent conversation.